dig.go

```diff
@@ -311,6 +331,28 @@ func (c *Container) getProviders(k key) []provider {
 	return providers
 }
 
+func (c *Container) missingNodes(params []*dot.Param) {
+	missing := make([]*dot.Result, len(params))
```

Comment:
Why do we need to map the `dot.Param` to a `dot.Result` here?

Reply:
In order to pinpoint which grouped node failed in the graph, we need the resulting failed nodes to be `dot.Result`s so the `GroupIndex` is included.